### ISSUE TITLE
Deserialization calls default setter before JsonSetter annotated method

### ISSUE THREAD

**Issue by OgmiosVoice** [FasterXML/jackson-databind#1710]
Upon deserialization Jackson `2.8.9` will also call the **setter of a property** before executing the `JsonSetter` **annotated method**:
```java
public class JsonSetterTest {
    static class Container {
        @JsonProperty("a")
        String aB;

        public String getAB() {
            return aB;
        }

//        @JsonIgnore
        public void setAB(final String aB) {
            this.aB = "setter";
        }

        @JsonSetter("a")
        public void setABJson(final String aB) {
            if (this.aB == null) {
                this.aB = aB;
            }
        }
    }

    @Test
    public void set() throws IOException {
        // GIVEN
        final ObjectMapper m = new ObjectMapper();
        final Container c = new Container();
        c.aB = "test";
        // WHEN
        final String json = m.writeValueAsString(c);
        Assert.assertEquals("{\"ab\":\"test\",\"a\":\"test\"}", json); // two properties are serialized
        final Container res = m.readValue(json, Container.class);
        // THEN
        Assert.assertEquals("test", res.aB); // fail
    }
}
```
I'm expecting on the other hand a **single method call**: the one annotated explicitly. Annotating the default setter with `JsonIgnore` will achieve this behaviour, yet this is an *undocumented "feature"* at this moment.

**Update**: quite annoying, this issue seems to be related with the **name of the field and of the getter/setter methods**: it only happens when the name of the field starts with a **single lowercase letter followed by an uppercase one** (e.g. `aB`), and this name is then capitalized in the name of the getter/setter methods (e.g. `getAB`, `setAB`) -- it works correctly with method names `getaB` and `setaB`.

**Comment by cowtowncoder** [comment]
It might help if you could either simplify the test (to include serialized json instead of relying on it getting written), or go step-by-step over how you expect things to work. There are some complexities in this case wrt casing of property names and complexity of correlating various fields, accessors (logic and order in which names are processed gets complex, in order to both minimize need for annotations and allow various ways of renaming, attaching/unattaching possible related methods and fields).

**Comment by OgmiosVoice** [comment]
I've started to extend the test according to your comment, when I've realized, that it is indeed only a casing problem: the current test serializes two variables: `a` (according to `JsonProperty`) and `ab` (because of `getAB`), and this is the reason behind calling both setters upon deserialization. (I must have failed to observe this also in my main project this test should have mimicked the behaviour of.)
And while this **capitalization** is illogical for me, it is the one defined by the [JavaBeans specification](http://www.oracle.com/technetwork/java/javase/documentation/spec-136004.html) (see also: [java.beans.Introspector.decapitalize(...)](http://docs.oracle.com/javase/8/docs/api/java/beans/Introspector.html#decapitalize-java.lang.String-)), so I must annotate all of these non-standard getters and setters too.
Thus I'm closing this bug as invalid. Sorry for the inconvenience.
